.github/workflows/github-pages-latest.yml

```
@@ -26,11 +26,17 @@ jobs:
         with:
           fetch-depth: 0 # Fetch all history for tags
 
-      - name: Get latest tag
-        id: get_latest_tag
+      - name: Ensure this is the latest tag
         run: |
           # Get all tags, sort them semantically, and pick the highest stable version
           LATEST_TAG=$(git tag --sort=-v:refname | grep -v - | head -n 1)
+          TRIGGER_TAG=$(git tag --points-at ${{ github.event.workflow_run.head_sha }} | head -n 1)
+
+          if [ "$TRIGGER_TAG" != "$LATEST_TAG" ]; then
+            echo "The triggered tag ($TRIGGER_TAG) is not the latest tag ($LATEST_TAG). Exiting."
+            exit 0
```

Check if this results in the intended outcome. We want to exit early without a status failure (exit 1) and not add `if` to every subsequent step. This doesn't seem to be supported 'yet' (https://github.com/orgs/community/discussions/82744)

---

I think you could get the workflow's `run_id` from the [`github` context](https://docs.github.com/en/actions/reference/workflows-and-actions/contexts#github-context) and use [`gh run cancel`](https://cli.github.com/manual/gh_run_cancel) to cancel it:

```suggestion
            gh run cancel ${{ github.run_id }}
```